### PR TITLE
kitchen-sync: add option to build with MariaDB

### PR DESCRIPTION
**Pull request by kuahyeow**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/kitchen-sync.rb [hunk 1]
```diff
@@ -19,6 +19,7 @@ class KitchenSync < Formula
```
Comment (fxcoudert) [inline Formula/kitchen-sync.rb]: Move to `depends_on "yaml-cpp"` without option

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by soundasleep** [comment]
+1 yes please

**Comment by ilovezfs** [comment]
You can already use the :mysql require with mariadb.

**Comment by kuahyeow** [comment]
Oh, I see. TIL !

**Pull request closed without merging**